File: backend/core-api/src/main/java/ai/govbiz/core/controller/SupportProgramController.java
package ai.govbiz.core.controller;

import ai.govbiz.core.dto.support.SupportProgramSearchResponse;
import ai.govbiz.core.service.SupportProgramSearchService;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequestMapping("/api/v1/support-programs")
public class SupportProgramController {

    private final SupportProgramSearchService searchService;

    public SupportProgramController(SupportProgramSearchService searchService) {
        this.searchService = searchService;
    }

    @GetMapping("/search")
    public SupportProgramSearchResponse search(
            @RequestParam String query,
            @RequestParam(defaultValue = "true") boolean acceptingOnly
    ) {
        return SupportProgramSearchResponse.from(searchService.search(query, acceptingOnly));
    }
}

File: backend/core-api/src/main/kotlin/ai/govbiz/core/controller/SupportProgramController.kt
package ai.govbiz.core.controller

import ai.govbiz.core.dto.support.SupportProgramSearchResponse
import ai.govbiz.core.service.SupportProgramSearchService
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RequestParam
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/api/v1/support-programs")
class SupportProgramController(
    private val searchService: SupportProgramSearchService,
) {

    @GetMapping("/search")
    fun search(
        @RequestParam query: String,
        @RequestParam(defaultValue = "true") acceptingOnly: Boolean,
    ): SupportProgramSearchResponse =
        SupportProgramSearchResponse.from(searchService.search(query, acceptingOnly))
}
